File: TestProj/Assets/HealthUI.js
var myPlayer : GameObject;
var myFadeElement : UI.Image; //Our blackUI thing.
function Start(){
	myPlayer = GameObject.FindGameObjectWithTag("Player");
}
function Update(){
	var healthAlpha = myPlayer.GetComponent("player").health *0.01;
	healthAlpha-=1;
	healthAlpha = Mathf.Abs(healthAlpha);
	if(healthAlpha > 0.9){
		healthAlpha = 0.9;
	}
	myFadeElement.color = new Color(0,0,0,healthAlpha);
}

File: TestProj/Assets/HealthUI.cs
using UnityEngine;
using System.Collections;
using UnityEngine.UI;

[System.Serializable]
public partial class HealthUI : MonoBehaviour
{
    public GameObject myPlayer;
    public Image myFadeElement; //Our blackUI thing.
    public virtual void Start()
    {
        myPlayer = GameObject.FindGameObjectWithTag("Player");
    }

    public virtual void Update()
    {
        float healthAlpha =  myPlayer.GetComponent<Player>().health * 0.01f;
        healthAlpha = (healthAlpha) - 1;
        healthAlpha = Mathf.Abs(healthAlpha);
        if (healthAlpha > 0.9f)
        {
            healthAlpha = 0.9f;
        }
        myFadeElement.color = new Color(0, 0, 0, healthAlpha);
    }

}